packages/patternfly-4/react-core/src/helpers/util.test.js

```diff
@@ -88,70 +88,86 @@ test('sideElementIsOutOfView Returns NONE when in view', () => {
   expect(sideElementIsOutOfView(container, element)).toBe(SIDE.NONE);
 });
 
-// TODO: Fix keyHandler Tests!!!
 // eslint-disable-next-line jest/no-disabled-tests
-xdescribe('keyHandler works on ApplicationLauncher', () => {
+describe('keyHandler works on ApplicationLauncher', () => {
   document.body.innerHTML = '<!doctype html><html><body></body></html>';
   const dropdownItems = [
-    <DropdownItem key="link" id="first">
-      Link
+    <DropdownItem key=" 1" id="first" component="button">
+      Action 1
     </DropdownItem>,
-    <DropdownItem key="action" id="second" component="button">
-      Action
+    <DropdownItem key="2" id="second" component="button">
+      Action 2
     </DropdownItem>,
-    <DropdownItem key="disabled link" id="third" isDisabled>
+    <DropdownItem key="3" id="third" component="button" isDisabled>
       Disabled Link
     </DropdownItem>
   ];
   const view = mount(<ApplicationLauncher dropdownItems={dropdownItems} isOpen />, {
     attachTo: document.getElementsByName('div')[0]
   });
 
-  const firstDropdownItem = view.find('#first').first();
-  const secondDropdownItem = view.find('#second').first();
+  const firstDropdownItem = view.find('#first button').first();
+  const secondDropdownItem = view.find('#second button').first();
 
   test('keyHandler advances forward', () => {
     firstDropdownItem.simulate('keydown', {
       key: 'ArrowDown',
       keyCode: KEY_CODES.ARROW_DOWN,
       which: KEY_CODES.ARROW_DOWN
     });
-    const focusElement = view.find(':focus').first();
-    expect(secondDropdownItem === focusElement).toBe(true);
+    const focusedElement = document.activeElement;
+    expect(focusedElement.getAttribute('id')).toEqual('second');
   });
 
   test('keyHandler regresses backward', () => {
-    secondDropdownItem.simulate('keydown', { key: 'ArrowUp', keyCode: KEY_CODES.ARROW_UP, which: KEY_CODES.ARROW_UP });
-    expect(firstDropdownItem === document.activeElement).toBe(true);
+    let focusedElement = document.activeElement;
+    expect(focusedElement.getAttribute('id')).toEqual('second');
```

Comment:
I would remove the initial expected focus item check from each of the tests. This makes each test depend on the previous test.

Reply:
Since there is no set-up or tear-down between each test case, they will depend on the previous test regardless. The following tests will fail if the DOM is not in the correct state. Maybe we should be initializing the DOM state from scratch on each test case?

Reply:
@jeff-phillips-18 what do you think?

Reply:
I think it's ok to just remove the init checks. The initial state is not handled by the utils and thus need not be tested here.